pkg/api/agentPoolOnlyApi/v20170831/types.go

```diff
@@ -94,6 +94,26 @@ const (
 	Migrating ProvisioningState = "Migrating"
 )
 
+// PoolUpgradeProfile contains pool properties:
+//  - kubernetes version
+//  - pool name (for agent pool)
+//  - OS type of the VMs in the pool
+//  - list of applicable upgrades
+type PoolUpgradeProfile struct {
+	KubernetesVersion string    `json:"kubernetesVersion"`
+	Name              string    `json:"name,omitempty"`
+	OSType            string    `json:"osType,omitempty"`
+	Upgrades          []*string `json:"upgrades,omitempty"`
+}
+
+// UpgradeProfile contains cluster properties:
```

Comment:
i'm not sure if this comment reflects this type correctly.

Reply:
addressed